app/src/main/java/com/vultisig/wallet/ui/components/v2/tokenitem/ChainSelectionItem.kt

```diff
@@ -165,23 +167,186 @@ internal fun GridPlus(
+
+@Preview
+@Composable
+fun PreviewRChainSelectionItem3() {
+    GridItem(
+        uiModel = TokenSelectionGridUiModel(
+            tokenSelectionUiModel = TokenSelectionUiModel.TokenUiPair(
+                left = TokenSelectionUiModel.TokenUiSingle(
+                    name = Coins.ThorChain.RUNE.ticker,
+                    logo = Coins.ThorChain.RUNE.chain.logo,
+                ),
+                right = TokenSelectionUiModel.TokenUiSingle(
+                    name = Coins.Bitcoin.BTC.ticker,
+                    logo = Coins.Bitcoin.BTC.chain.logo,
+                )
+            ),
+            isChecked = true
+        )
+    )
+}
+
+
+@Composable
+private fun TokenUiGridLogo(
+    tokens: TokenSelectionUiModel.TokenUiPair,
+    space: Float = 4f,
+) {
+
+    val rightToken = tokens.right
+    val leftToken = tokens.left
+    val halfSpace = space / 2
+
+    Box {
+        TokenUiGridLogo(
+            token = leftToken,
+            modifier = Modifier
+                .clip(
+                    shape = HalfClipper(
+                        clipSide = ClipSide.Left,
+                        offset = halfSpace
+                    )
+                )
+        )
+
+        TokenUiGridLogo(
+            token = rightToken,
+            modifier = Modifier
+                .clip(
+                    shape = HalfClipper(
+                        clipSide = ClipSide.Right,
+                        offset = halfSpace
+                    )
+                )
         )
+    }
+
+}
+
+
+@Composable
+private fun TokenUiGridLogo(
+    modifier: Modifier = Modifier,
+    size: Dp = 28.dp,
+    token: TokenSelectionUiModel.TokenUiSingle,
+) {
+
+    Box(
+        modifier = modifier
+    ) {
+        SubcomposeAsyncImage(
+            model = token.logo,
+            contentDescription = null,
+            modifier = Modifier.size(size),
+            loading = {
+                V2Loading()
+            },
+            error = {
+                Box(
+                    modifier = Modifier.size(size),
+                    contentAlignment = Alignment.Center
+                ) {
+                    Text(
+                        token.name.firstOrNull()?.toString().orEmpty(),
+                        color = Theme.colors.text.primary,
+                        style = Theme.brockmann.body.m.medium
+                    )
+                }
+            }
+        )
+    }
+
+
+}
+
+@Composable
+private fun TokenUiGridName(
+    token: TokenSelectionUiModel.TokenUiSingle,
+) {
+    Text(
+        text = token.name,
+        style = Theme.brockmann.supplementary.caption,
+        color = Theme.colors.text.primary,
+        modifier = Modifier
+            .widthIn(max = 74.dp),
+        maxLines = 1,
+        overflow = TextOverflow.Ellipsis
+    )
+}
+
+@Composable
+private fun TokenUiGridName(
+    tokens: TokenSelectionUiModel.TokenUiPair,
+    mapper: (TokenSelectionUiModel.TokenUiPair) -> String = { "${it.left.name}/${it.right.name}" },
```

Comment:
how this is meant to be used? It would be nice if you can have preview of full dialog 👍🏻